### PR TITLE
Land enableCustomElementPropertySupport for React 19

### PR DESCRIPTION
**Pull request by alunyov**
We've rolled out this flag internally on WWW. This PR removed flag `enableCustomElementPropertySupport`

Test plan:
 -- `yarn test`

### CODE REVIEW

**Comment by kassens** [comment]
Can we just delete the flag?

**Comment by alunyov** [comment]
>  Can we just delete the flag?

Yeah, I guess we could do that.

**Comment by sebmarkbage** [comment]
There's a known issue with this flag here: https://github.com/facebook/react/issues/27286

**Comment by alunyov** [comment]
Rebased on top of @gnoff fix for #27286

**Review by poteto — APPROVED** [review]

File: packages/react-dom-bindings/src/client/ReactDOMComponent.js [hunk 1]
```diff
@@ -2213,30 +2200,28 @@ function diffHydratedCustomComponent(
...
+          console.error(
+            'Assignment to read-only property will result in a no-op: `%s`',
+            propKey,
           );
-          warnForPropDifference('className', serverValue, value);
-          continue;
         }
+        continue;
+      }
+      // Fall through
```
Comment (kassens) [inline packages/react-dom-bindings/src/client/ReactDOMComponent.js:2213]: Remove, this doesn't fall through.

**Comment by michaelwarren1106** [comment]
just pinging this thread to see if there are any barriers to getting this merged? This PR is a clear signal of web component support in react and a lot of web components advocates are eagerly awaiting :)

**Comment by gnoff** [comment]
@michaelwarren1106 our plan is to include this in our next stable release which is being put together now. It will be merged closer to when we are ready to publish

**Comment by klebba** [comment]
@gnoff While your comment above gives me some hope, I still find myself wondering if I actually understand what to expect in the next stable release. Is there a roadmap published with the contents of the next stable release somewhere? I could not find anything myself; I have some older information about [what you're working on](https://react.dev/blog/2023/03/22/react-labs-what-we-have-been-working-on-march-2023) but I don't consider everything I read to be a commitment or a comprehensive list (and support for custom element support is never mentioned). 

To be honest monitoring GitHub threads (like this one) for the last several years has been frustrating and confusing. *Will the React team please state plainly what to expect regarding support for custom elements in a blog post here: [react.dev/blog](https://react.dev/blog)?*

As evidence that there is tremendous appetite for this content I will mention this relates to the [#2 most discussed issue](https://github.com/facebook/react/issues?q=is%3Aissue+sort%3Acomments-desc) and #1 most hearted issue (it's [the same](https://github.com/facebook/react/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-heart-desc)) in the list of public issues.

What I *think* I can expect is that the next major release will show all 16 basic & advanced tests on (custom-elements-everywhere.com)[https://custom-elements-everywhere.com] will be passing in React 19. Today this is true for `React^0.0.0-experimental-3ddbedd05-20220719` which I am really happy to see. However without an announcement in an official public channel I must still hold my applause (for now).

Sorry for the rant; I do appreciate the React team's efforts! Also 👋🏻 @poteto — it's good to see a familiar face working on this :)

**Comment by rickhanlonii** [comment]
@klebba we're finalizing the release and then we'll be able to share what's in it. For custom elements, since this PR turns the flag on from `experimental`, they will ship with the PR.

**Comment by klebba** [comment]
Hi @rickhanlonii — thanks for your reply. I'll reiterate what I'm asking, I don't intend to be controversial and only want to know about this subject:

> this PR turns the `enableCustomElementPropertySupport` flag on from experimental

What exactly does this flag do? What does it not do? I still hope the React team will soon state plainly what to expect regarding support for custom elements in a blog post here: [react.dev/blog](https://react.dev/blog).

To be clear, I'm not asking:

* For anything to be merged
* For any schedule commitments to be made

I'm recommending:

* Official communication about planned future support for custom elements in React
* Explicit enumeration of what's in / what's out / unknown

So that I can have:

* Ability to un-watch #11347
* Ability to inform expensive forward looking business decisions regarding React
* An official statement I can share with colleagues and executives
* Confidence that I won't be [disappointed](https://github.com/facebook/react/issues/11347#issuecomment-713175344) and stuck again-again (no blame or fault intended, but I was still disappointed and stuck):

Screenshot 2024-01-18 at 9 55 13 AM

My dream would be to read a blog post on [react.dev/blog](https://react.dev/blog) along these lines:

> **Plan for custom elements support in next stable release**
>
> TL;DR — React 19 will now be more compatible with and have direct support for custom elements. Developers can expect that all tests on custom-elements-everywhere.com will pass by default going forward. Release date still to be announced.
> 
> Special thanks to @robdodson for championing this cause!

Thanks for your time and effort!

**Comment by rickhanlonii** [comment]
@klebba I posted an update in https://github.com/facebook/react/issues/11347#issuecomment-1899140345, we'll share updates on the blog when we share more about the next release.

**Comment by klebba** [comment]
Thank you @rickhanlonii — I really appreciate the clarity!

**Comment by rickhanlonii** [comment]
Rebased on main

File: packages/react-dom-bindings/src/client/ReactDOMComponent.js (hunk 1 above)
Comment (rickhanlonii) [reply packages/react-dom-bindings/src/client/ReactDOMComponent.js:2291]: Nice catch, fixed
